src/realm/dictionary.cpp

```diff
@@ -1031,6 +1041,12 @@ void Dictionary::to_json(std::ostream& out, size_t link_depth, JSONOutputMode ou
             Lst<Mixed> list(parent, 0);
             list.to_json(out, link_depth, output_mode, fn);
         }
+        else if (val.is_type(type_Set)) {
```

Comment:
Why not use same constructs as above?

Reply:
Actually, it turns out that all I need is `CollectionParent::Index`